File: classic puzzle-hard/the-greatest-number.py
#https://www.codingame.com/training/hard/the-greatest-number
#THE GREATEST NUMBER
import re
n=int(input())
a=input().split()
s=re.findall(r'[0-9]',''.join(a))
num=''
if '-' in a:
    num='-'+''.join(sorted(s,key=lambda x:x))
    if '.' in a:
        num=num[:2]+'.'+num[2:]
else:
    num=''.join(sorted(s,key=lambda x:-int(x)))
    if '.' in a:
        num=num[:-1]+'.'+num[-1:]
if float(num).is_integer():
    if '.' in num: print(int(num[:num.index('.')]))
    else: print(int(num))
else:
    print(float(num))

File: classic puzzle-medium&above/the-greatest-number.js
//https://www.codingame.com/training/hard/the-greatest-number
//THE GREATEST NUMBER
const N = parseInt(readline());
const s = readline();
var a=s.match(/[0-9]/gm);
var num;
if(s.includes('-')){
    num='-'+a.sort((a,b)=>a-b).join('');
    if(s.includes("."))
        num=num.slice(0,2)+"."+num.slice(2,)
}else{
    num=a.sort((a,b)=>b-a).join('');
    if(s.includes("."))
        num=num.slice(0,-1)+"."+num.slice(-1)
}
console.log(parseFloat(num).toString());
